stm32-modules/heater-shaker/firmware/heater_task/heater_hardware.c

```diff
@@ -284,6 +286,33 @@ void HAL_ADC_ConvCpltCallback(ADC_HandleTypeDef* hadc) {
     }
 }
 
+bool heater_hardware_set_offsets(struct writable_offsets* to_write) {
+    FLASH_EraseInitTypeDef pageToErase = {.TypeErase = FLASH_TYPEERASE_PAGES, .PageAddress = OFFSETS_PAGE_ADDRESS, .NbPages = 1};
+    uint32_t pageErrorPtr = 0; //pointer to variable  that contains the configuration information on faulty page in case of error
+    uint32_t ProgramAddress1 = OFFSETS_PAGE_ADDRESS + (offsetof(struct writable_offsets, const_b) * 8); //offsetof returns bytes
```

Comment:
wait this doesn't quite click with me - there's no way program address is a bit address. Shouldn't this be a `/ 8`?

Reply:
ProgramAddressX is in bytes, so we can just use offsetof(). Got rid of the multiply/divide by 8